src/figuresClasses.ts

```diff
@@ -1,11 +1,102 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Color;
+
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
+
+  constructor(color: Color, sideA: number, sideB: number, sideC: number) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('any length is <= 0');
+    }
+
+    if (
+      sideA >= sideB + sideC ||
+      sideB >= sideA + sideC ||
+      sideC >= sideA + sideB
+    ) {
+      throw new Error(
+        'the longest side of a triangle is >= than a sum of two others',
+      );
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.sideA = sideA;
+    this.sideB = sideB;
+    this.sideC = sideC;
+  }
+
+  getArea(): number {
+    const s = (this.sideA + this.sideB + this.sideC) / 2;
+    const area = Math.sqrt(
+      s * (s - this.sideA) * (s - this.sideB) * (s - this.sideC),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle';
 
-export class Circle implements Figure {}
+  color: Color;
 
-export class Rectangle implements Figure {}
+  radius: number;
+
+  constructor(color: Color, radius: number) {
+    if (radius <= 0) {
+      throw new Error('radius must be > 0');
+    }
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
```

Comment:
The `getArea` method should not accept any parameters as all necessary values are available as object fields.